server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -164,6 +164,19 @@ public void writeTo(StreamOutput out) throws IOException {
      */
     public abstract long nextRoundingValue(long value);
 
+    /**
+     * How "offset" this rounding is from the traditional "start" of the period.
+     * @deprecated We're in the process of abstracting offset *into* Rounding
+     *             so keep any usage to migratory shims
+     */
+    @Deprecated
+    public abstract long offset();
```

Comment:
What's the plan to migrate away from this? I'm not sure I understand the benefit to adding a deprecated method now that we intend to remove in the next PR, as opposed to just removing the need for it in this PR.

Reply:
I added this to cut the PR in half, basically. I'll remove it in a followup. I believe that follow up with have serialization changes so it'll *feel* pretty different than this PR.

Reply:
Seems reasonable, thanks for explaining!